### DISCUSSION TITLE
Multifactor authentication with Sharepoint Online

### DISCUSSION THREAD

**Discussion by zmajcekri** [pnp/pnpframework#839] [General]
Is there an example of how to use this framework to get a client context of a sharepoint online site using MFA ?

**Comment by pkbullock** [comment]
Azure AD apps with the application permissions do not need MFA. Is there a particular authentication scenario you are looking for?

**Comment by henrik0805** [comment]
Do you mean an ability to show web-component where users are presented with the usual MS Log-in procedure?
BrowserHelper.cs can be used, although I have a experienced that SPO does not like the fact that EI11 is used as the browser-controller. I have rewritten the BrowserHelpder.cs to use WebView2 (i.e. EDGE) instead.
But, you get the Context like this:
 Context = BrowserHelper.GetWebLoginClientContext([Site], true/false);
